cdrs-tokio/src/cluster/session.rs

```
@@ -817,7 +818,11 @@ impl<
             version,
         );
 
-        let control_connection_handle = tokio::spawn(control_connection.run());
+        let (init_complete_sender, init_complete_receiver) = tokio::sync::oneshot::channel();
+        let control_connection_handle = tokio::spawn(control_connection.run(init_complete_sender));
+        if init_complete_receiver.await.is_err() {
+            tracing::error!("session control connection died without registering init complete")
```

We should return an error in such case.